python/hidet/runtime/compiled_graph.py

```diff
@@ -277,6 +277,33 @@ def run_async(self, inputs):
         ret: List[hidet.Tensor]
             The output tensors.
         """
+        if hidet.option.runtime_check():
+            symbol_map = {}
+            for i, (traced, new) in enumerate(zip(self.meta.inputs, inputs)):
+                if hidet.ir.data_type(traced.dtype) != hidet.ir.data_type(new.dtype):
+                    raise RuntimeError(
+                        f"dtype mismatch at arg {i} between original: {traced.dtype} and new: {new.dtype}"
+                    )
+                traced_shape = traced.shape
+                concrete_shape = new.shape
+                if len(traced_shape) != len(concrete_shape):
+                    raise RuntimeError(
+                        f"Rank of input {i} not equal to original. ({len(concrete_shape)} vs. {len(traced_shape)})"
+                    )
+                for j, (orig_shape, new_shape) in enumerate(zip(traced_shape, concrete_shape)):
```

Comment:
```suggestion
                for j, (expected_size, actual_size) in enumerate(zip(traced_shape, concrete_shape)):
```
The shape usually means all dimension sizes